### DISCUSSION TITLE
Adding more granular permissions

### DISCUSSION THREAD

**Discussion by soof-golan** [community/community#18360] [Other Feature Feedback, Questions, & Ideas]
# TL;DR

3rd party tools (like [Streamlit](https://streamlit.io/)) request full admin access to all of your account 😕 
Github's permission system is missing a partial access feature to let you control these permissions.

# Concrete example

Streamlit asks for admin permissions for [all my repos](https://github.com/streamlit/streamlit/issues/4344), even the ones that it doesn't need

# Possible solutions

## Selective Repository Access

Github gives the 3rd party service access to [whichever repos I pick], but the 3rd party service still thinks it got access to all of my repos

## Virtual Proxy Accounts

Github should implement something similar to Apple's _Hide My Email_ feature, but with repositories.
When signing in with a Github account, provide an option to create a virtual account, with limited permission (and limited repo access) to the main account. The 3rd party would never be aware of this.
